### PR TITLE
Add new build profile

### PR DESCRIPTION
**Pull request by ksclarke**
* Add build profile in the release GA
* Update dependencies and base Docker image for security fixes

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
           gpg_passphrase: ${{ secrets.BUILD_PASSPHRASE }}
           nexus_username: ${{ secrets.SONATYPE_USERNAME }}
           nexus_password: ${{ secrets.SONATYPE_PASSWORD }}
-          maven_profiles: release
+          maven_profiles: release,optional-env-vars
```
Comment (markmatney) [inline .github/workflows/release.yml:58]: Why this change?

Comment (ksclarke) [reply .github/workflows/release.yml:58]: Was getting failed "release" builds (https://github.com/UCLALibrary/hauth/actions/runs/4246752618/jobs/7384531491), but the regular "build" was working so I looked at the difference between the release.yml and build.yml. The build.yml had been updated with this profile but the release.yml had not been. I added it to the release.yml and it worked. It looks like it defines some cookie/token defaults for the tests, which also run as a part of the release build.

**Review by markmatney — APPROVED** [review]

**Review by DRickard — APPROVED** [review]
